src/quickpicks/schemaRegistries.ts

```diff
@@ -32,44 +32,19 @@ export async function schemaRegistryQuickPickWithViewProgress(): Promise<
 }
 
 /**
- * Create a quickpick to let the user choose a Schema Registry, named by environment.
+ * Execute a quickpick to let the user choose a Schema Registry, named by (ccloud or local) environment.
+ * If the user has selected a schema registry in the schemas view, it will be the default selection.
  *
  * @returns The selected Schema Registry, or undefined if none was selected.
  */
 export async function schemaRegistryQuickPick(): Promise<SchemaRegistry | undefined> {
-  const localSchemaRegistries: LocalSchemaRegistry[] = [];
-  let cloudSchemaRegistries: CCloudSchemaRegistry[] = [];
+  const registriesByConnectionID: Map<string, SchemaRegistry[]> =
+    await getRegistriesByConnectionID();
 
-  // schema registries are a coarse resource, so ensure they are loaded before proceeding
-  const ccloudLoader = ResourceLoader.getInstance(CCLOUD_CONNECTION_ID);
-  await ccloudLoader.ensureCoarseResourcesLoaded();
-
-  // first we grab all available (local+CCloud) Schema Registries
-  let localGroups: LocalResourceGroup[] = [];
-  let cloudGroups: CCloudSchemaRegistryByEnv | null = new Map<string, CCloudSchemaRegistry>();
-  [localGroups, cloudGroups] = await Promise.all([
-    getLocalResources(),
-    getResourceManager().getCCloudSchemaRegistries(),
-  ]);
-  localGroups.forEach((group) => {
-    if (group.schemaRegistry) {
-      localSchemaRegistries.push(group.schemaRegistry);
-    }
-  });
-
-  // list all Schema Registries for all CCloud environments for the given connection; to be separated
-  // further by environment in the quickpick menu below
-  let cloudEnvironmentIds: string[] = [];
-  if (hasCCloudAuthSession()) {
-    cloudGroups.forEach((schemaRegistry, envId) => {
-      cloudEnvironmentIds.push(envId);
-      cloudSchemaRegistries.push(schemaRegistry);
-    });
-  }
+  const localSchemaRegistries = registriesByConnectionID.get(LOCAL_CONNECTION_ID)!;
+  const ccloudSchemaRegistries = registriesByConnectionID.get(CCLOUD_CONNECTION_ID)!;
```

Comment:
Are these still the right return types as before?
```suggestion
  const localSchemaRegistries: LocalSchemaRegistry[] = registriesByConnectionID.get(LOCAL_CONNECTION_ID)!;
  const ccloudSchemaRegistries: CCloudSchemaRegistry[] = registriesByConnectionID.get(CCLOUD_CONNECTION_ID)!;
```